File: Data Structures & Algorithms/longest-common-prefix/submission-0.cpp
class Solution {
public:
    string longestCommonPrefix(vector<string>& strs) {
        sort(strs.begin(), strs.end());
        string res = "";
        int i = 0, n = strs.size(), m = strs[0].size();
        while (i < m && strs[0][i] == strs[n-1][i]) {
            res += strs[0][i];
            i++;
        }
        return res;
    }
};

File: Data Structures & Algorithms/sliding-window-maximum/submission-0.py
class Solution:
    def maxSlidingWindow(self, nums: List[int], k: int) -> List[int]:
        deq = deque()
        res = []
        l = 0
        for r in range(len(nums)):
            # maintaining window size
            if r - l + 1 > k:
                if nums[l] == deq[0]:
                    deq.popleft()
                l += 1
            
            # maintaining monotonic decreasing order in deque
            # (equal allowed) 
            while deq and deq[-1] < nums[r]:
                deq.pop()
            
            deq.append(nums[r])

            if r - l + 1 == k:
                res.append(deq[0])
            
        return res